input_definition.go

```diff
@@ -315,3 +317,61 @@ func (i *InputDefinition) ValidateAction(action *internal.InputDefinitionAction)
 	}
 	return nil
 }
+
+// HandleAction Process the input data with its action and return a bit to be imported later
+// Note: if the Bit should not be set then nil is returned with no error
+// From the JSON marshalling the possible types are: float64, boolean, string
+// TODO handle Timestams
+func HandleAction(a *Action, value interface{}, colID uint64) (*Bit, error) {
+	var err error
+	var bit Bit
+	bit.ColumnID = colID
+
+	switch a.ValueDestination {
+	case Mapping:
+		v, ok := value.(string)
+		if !ok {
+			return nil, fmt.Errorf("Mapping value must be a string %v", value)
+		}
+		bit.RowID, ok = a.ValueMap[v]
+		if !ok {
+			return nil, fmt.Errorf("Value %s does not exist in definition map", v)
+		}
+	case SingleRowBool:
+		switch value.(type) {
```

Comment:
I think user must pass bool here. Like if definition as following:
```
{
					"name": "withPet",
					"actions": [
						{
							"frame": "add-ons",
							"valueDestination": "single-row-boolean",
							"rowID": 100
						}
					]
				}
```
					
the JSON input must be:
```
[{"with-pet": true}]
```
So that we just need to assert